pallets/avn-oracle/src/lib.rs

```diff
@@ -529,33 +518,55 @@ pub mod pallet {
         }
 
         fn format_rates(prices_json: Vec<u8>) -> Result<Rates, DispatchError> {
-            let prices_str = String::from_utf8_lossy(&prices_json);
-            let prices: Value = serde_json::from_str(&prices_str)
+            let prices: Value = serde_json::from_slice(&prices_json)
                 .map_err(|_| DispatchError::Other("JSON Parsing Error"))?;
 
-            let mut formatted_rates: Vec<(CurrencySymbol, U256)> = Vec::new();
+            let mut formatted_rates: Vec<(Currency, U256)> = Vec::new();
 
             if let Some(rates) = prices.as_object() {
-                if let Some((symbol, rate_value)) = rates.iter().next() {
+                for (currency, rate_value) in rates {
```

Comment:
if you want to avoid nesting `if`'s you could replace 
```
if let Some(rate) = rate_value.... {
..
}
```
 with 
`let rate = rate_value.as_f64().ok_or(Error::<T>::InvalidRateFormat)?;`